### PR TITLE
SBOMit Sandbox Project Application

### PR DESCRIPTION
**Pull request by idunbarh**
The SBOMit project is seeking sandbox maturity level entry into the OpenSSF.

SBOMit was presented to the Security Tooling WG on 8/15/23 and the slides are available at ...

- https://docs.google.com/presentation/d/1l3dPqw67uf9VQ4yJEbhZkjhHhhhEYpjGdrWclmdTJFg/edit?usp=sharing

We believe we meet OpenSSF Sandbox requirements. 

- _Projects must have a minimum of two maintainers with different organization affiliations._
  - Project maintainers include
    - Justin Cappos, NYU, @justincappos
    - Ian Dunbar-Hall, Lockheed Martin, @idunbarh
    - Cole Kennedy, TestifySec, @colek42
    - Marina Moore, NYU, @mnm678
    - Trishank Kuppusamy, Datadog, @trishankatdatadog
- _Projects must be aligned with the OpenSSF mission and either be a novel approach for existing areas or address an unfulfilled need. It is expected that the initial code needed for an OpenSSF WG to work be kept within their repository and will not function as a project in its own right. Should initial WG code grow and mature that it warrants its own Project status, then it is subject to Sandbox entry requirements. It is preferred that extensions of existing OpenSSF projects collaborate with the existing project rather than seek a new project._
  - Software Bills of Materials (SBOMs) often suffer from inaccuracy, primarily due to a retrospective approach to software analysis that fails to accurately track components. This challenge is becoming more pressing with the rise of sophisticated supply chain attacks requiring advanced technology and robust frameworks like the Secure Software Development Framework (SSDF) for comprehensive security.
- _If contributing an existing Project to the OpenSSF, the contribution must undergo license and IP due diligence by the Linux Foundation (LF)._
  - See [issue requesting IP review](https://github.com/ossf/tac/issues/191)

__Remaining Actions__

- [x] Complete [IP Review](https://github.com/ossf/tac/issues/191)
- [ ] Security Tools WG support confirmation

SBOMit Links

- https://github.com/sbomit
- https://sbomit.dev/

SBOMit is licensed as CC-BY-4.0.

- https://github.com/SBOMit/specification/blob/main/LICENSE.md
- https://github.com/SBOMit/website/blob/main/LICENSE.md

### CODE REVIEW

**Comment by mlieberman85** [comment]
I read through what's currently there for the specification, and one big suggestion I have is an example, even if it's a contrived hello world example would be useful to really understand how SBOMit intends to solve the problem posed in the specification and presentation.

**Comment by mlieberman85** [comment]
(starting the voting thread) +1 Binding from me.

**Comment by SecurityCRob** [comment]
voting - +1 for me to adopt based on reaction to feedback & adjustment of license (thanks!)

**Comment by steiza** [comment]
+1 for me - the project is in a early phase, but that's what the sandbox step is for!

**Comment by lehors** [comment]
As I explained multiple times in various places: under our current governance structure this cannot be brought in as a project. Because it is not primarily focused on producing code, it ought to be categorized as a SIG (for which there is no sandbox stage).
But I'm otherwise fine with bringing this work into OpenSSF.

**Comment by torgo** [comment]
+1 from me.

**Comment by bobcallaway** [comment]
+1 from me

File: process/project-lifecycle-documents/SBOMit_sandbox_stage.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+Non-Goals:
+ * Picking a winning SBOM format (SPDX, CycloneDX, etc.)
+ * Recursing into components like the packages inside of a container image when the build process does not otherwise do so.
+ * Knowing that an individual action is actually a good security practice  
+ * Assertions about the quality of the implementation of the tool / security processes describing how the SBOM or artifact came to exist
+
+
+
+
+### IP policy and licensing due dilligence
```
Comment (jsoref) [inline process/project-lifecycle-documents/SBOMit_sandbox_stage.md:43]:
```suggestion
### IP policy and licensing due diligence
```

**Comment by jsoref** [comment]
Assuming the other items are correctly spelled, can you try following the instructions in https://github.com/ossf/tac/actions/runs/6218203711?pr=192#summary-16874345424 ? (If they don't make sense, please let me know)

**Comment by JustinCappos** [comment]
I'm finding a bit ironic that the OpenSSF is asking me to curl and run a perl script from the Internet...  😃 

> To accept ✔️ these unrecognized words as correct, run the following commands
> ... in a clone of the [git@github.com:SBOMit/tac.git](https://github.com/SBOMit/tac.git) repository
> on the main branch ([ℹ️ how do I use this?](https://github.com/check-spelling/check-spelling/wiki/Accepting-Suggestions)):
> 
> curl -s -S -L 'https://raw.githubusercontent.com/check-spelling/check-spelling/v0.0.21/apply.pl' |
> perl - 'https://github.com/ossf/tac/actions/runs/6218203711/attempts/1'

I'm not super conversant in Perl and I'm about to head to bed.  The flagged items are actually spelled correctly, if someone else from the SBOMit group has a moment to sanity check what is happening and then tell the spellchecker to ignore these issues.

**Comment by jsoref** [comment]
You can manually add the items to https://github.com/ossf/tac/blob/main/.github/actions/spelling/expect.txt if you're more comfortable w/ that. And https://github.com/ossf/tac/blob/main/.github/actions/spelling/advice.md can be adjusted to provide instructions to do that.

**Comment by JustinCappos** [comment]
Okay, I've submitted a PR with those words added.  https://github.com/ossf/tac/pull/200

**Comment by jsoref** [comment]
@JustinCappos  You kind of want to do something like https://github.com/SBOMit/tac/compare/main...JustinCappos:tac:patch-2 -- i.e. creating the PR from your repo into the SBOMit repo.

check-spelling will suggest deleting things that aren't needed, so those would be orphaned and reaped by the next person who followed its instructions.

**Comment by mlieberman85** [comment]
> > 6 of 7 TAC members voted to approve, with 1 not responding. Arnaud brings up good points that we need to have a spec-focused group entry in our governance, but that should not block this effort.
> 
> 
> 
> Thanks! So is this pr ready to merge?

Yep just need to fix the merge conflicts and it's ready to go

**Comment by hythloda** [comment]
> > > 6 of 7 TAC members voted to approve, with 1 not responding. Arnaud brings up good points that we need to have a spec-focused group entry in our governance, but that should not block this effort.
> > 
> > 
> > Thanks! So is this pr ready to merge?
> 
> Yep just need to fix the merge conflicts and it's ready to go

I tried to resolve the conflicts but I don't have write permission on this branch to help :(

**Comment by hythloda** [comment]
@jsoref is the spelling workflow broken or something in the file?

**Comment by jsoref** [comment]
@hythloda unfortunately check-spelling is looking at files as a whole, it found these items in the files changed in this PR:

[`CRob`](https://github.com/ossf/tac/blame/aa5f07083f5133e6faf45475ca058e5d851b9414/README.md#L30)
[`Roadmap`](https://github.com/ossf/tac/blame/aa5f07083f5133e6faf45475ca058e5d851b9414/process/project-lifecycle-documents/SBOMit_sandbox_stage.md#L59)
[`Sigstore`](https://github.com/ossf/tac/blame/aa5f07083f5133e6faf45475ca058e5d851b9414/README.md#L80)
[`TUF`](https://github.com/ossf/tac/blame/aa5f07083f5133e6faf45475ca058e5d851b9414/README.md#L75)

The reason it cares about `TUF` and not  `tuf` is because there's a [pattern](https://github.com/ossf/tac/blob/f11de8b763f9465c15a0a6769957b7f2310898f9/.github/actions/spelling/patterns.txt#L6) that means that it won't see the [url that contains `tuf`](https://github.com/ossf/tac/blame/aa5f07083f5133e6faf45475ca058e5d851b9414/README.md#L75) -- and it has no idea if `tuf` is used throughout the rest of the repository because it isn't checking the other files.

I just released a version last week which would have told you which lines it found each item in (which should have helped things a lot) and which files it was checking (which would have improved things a little).

I need to think about the logic of the _only changed_ files feature (it exists for this sort of repository use case, but I rarely use it) -- I suspect the advice it gives should be to add items into `allow.txt` instead of into `expect.txt` (a distinction here is that `allow.txt` doesn't do any pruning, and doesn't have any case tolerance, so it wouldn't have promised that the presence of `tuf` meant that `TUF` was ok). I have a PR drafted to upgrade this repository to the version I just released, but in testing it, I realized that it wasn't offering `advice.md` at all, which isn't great, so I was thinking about what to do (the easiest thing here is to change the `advice.md` file to suggest using `allow.txt` instead...)

But basically, the goal here is for it to ask people if they're ok with the words that are used, if they are, the check isn't mandatory and once it merges, future PRs that don't change those files won't be asked about those items.

Oh..., there is a workflow change we could/should make which is adding `quit_without_error: 1` to these two `with` blocks:
https://github.com/ossf/tac/blob/aa5f07083f5133e6faf45475ca058e5d851b9414/.github/workflows/spelling.yml#L131
https://github.com/ossf/tac/blob/aa5f07083f5133e6faf45475ca058e5d851b9414/.github/workflows/spelling.yml#L148

That'd result in you not seeing a ❌ but instead getting the ✅ .

So, long story short:
1. to get a ✅ here, add the four words to `allow.txt` or `expect.txt` (it won't care, I'd favor the former)
2. to get ❌s to stop appearing going forward, we'd want to copy `quit_without_error: 1` to those two other `with:` blocks.
